interpreter/cling/include/cling/Interpreter/RuntimePrintValue.h

```diff
@@ -247,72 +247,21 @@ namespace cling {
 
   // Tuples
   template <class... ARGS>
-  inline std::string printValue(std::tuple<ARGS...> *);
-
-  namespace collectionPrinterInternal {
-    // We loop at compile time from element 0 to element TUPLE_SIZE - 1
-    // of the tuple. The running index is N which has as initial value
-    // TUPLE_SIZE. We can therefore stop the iteration and account for the
-    // empty tuple case with one single specialisation.
-    template <class TUPLE,
-              std::size_t N = std::tuple_size<TUPLE>(),
-              std::size_t TUPLE_SIZE = std::tuple_size<TUPLE>()>
-    struct tuplePrinter {
-      static std::string print(TUPLE *t)
-      {
-        constexpr std::size_t elementNumber = TUPLE_SIZE - N;
-        using Element_t = decltype(std::get<elementNumber>(*t));
-        std::string ret;
-        if (elementNumber)
-           ret += ", ";
-        ret += cling::printValue(&std::get<elementNumber>(*t));
-        // If N+1 is not smaller than the size of the tuple,
-        // reroute the call to the printing function to the
-        // no-op specialisation to stop recursion.
-        constexpr std::size_t Nm1 = N - 1;
-        ret += tuplePrinter<TUPLE, Nm1>::print((TUPLE *)t);
-        return ret;
-      }
-    };
-
-    // Special case: no op if last element reached or empty tuple
-    template <class TUPLE, std::size_t TUPLE_SIZE>
-    struct tuplePrinter<TUPLE, 0, TUPLE_SIZE>
-    {
-      static std::string print(TUPLE *t) {return "";}
-    };
-
-    template <class T>
-    inline std::string tuplePairPrintValue(T *val)
-    {
-      std::string ret("{ ");
-      ret += collectionPrinterInternal::tuplePrinter<T>::print(val);
-      ret += " }";
-      return ret;
-    }
-  }
-
-  template <class... ARGS>
-  inline std::string printValue(std::tuple<ARGS...> *val)
-  {
-    using T = std::tuple<ARGS...>;
-    if (std::tuple_size<T>::value == 0)
+  inline std::string printValue(std::tuple<ARGS...>* val) {
+    const auto nargs = std::tuple_size<std::tuple<ARGS...>>::value;
+    if (nargs == 0)
       return valuePrinterInternal::kEmptyCollection;
-    return collectionPrinterInternal::tuplePairPrintValue<T>(val);
-  }
-
-  template <class... ARGS>
-  inline std::string printValue(std::pair<ARGS...> *val)
-  {
-    using T = std::pair<ARGS...>;
-    return collectionPrinterInternal::tuplePairPrintValue<T>(val);
-  }
 
-  namespace collectionPrinterInternal {
-    // Keep this last to allow picking up all specializations above.
-    template <typename T> std::string printValue(const T* V, const void*) {
-      return cling::printValue(V);
-    }
```

Comment:
Removing this function may be behind the errors seen all over the place, e.g. https://lcgapp-services.cern.ch/root-jenkins/job/root-pullrequests-build/184557/testReport/projectroot/runtutorials/tutorial_dataframe_df024_Display/

This was practically the final specialization available for the forward declaration at https://github.com/root-project/root/blob/bd038661ed8fcaa7ae3b6529cbfb2a0267f7eb8d/interpreter/cling/include/cling/Interpreter/RuntimePrintValue.h#L159

But now, all the implementations of `collectionPrinterInternal::printValue` are gone, so we get the unresolved symbol issue. 

The real question is whether we actually need it or not, maybe yes to still allow recursive printing of container items, to be investigated

Reply:
Thanks @vepadulano. It really seems I removed too much code by mistake. I rebased and refreshed the PR. Let's see what the CI has to say about the new code.